dev/NavigationView/NavigationView.cpp

```diff
@@ -1744,7 +1744,7 @@ void NavigationView::UpdatePaneButtonsWidths()
 
     if (auto&& backButton = m_backButton.get())
     {
-        backButton.Width(newButtonWidths);
+        backButton.Width(newButtonWidths - 8);
```

Comment:
Changes in margin widths that will break the older styling. Is there some way to detect the app's version much like the .xaml files? @licanhua @ranjeshj

Reply:
@karenbtlai check this file, there are a lot of helper functions to determine [link](https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/dll/SharedHelpers.cpp)
![image](https://user-images.githubusercontent.com/21356912/105646388-8ebdcf80-5e54-11eb-87ee-cfc6bfbc9e59.png)

Reply:
@beervoley By version, I don't mean the OS version, but the `Version` property defined with ` <controls:XamlControlsResources Version="Latest"/>` in `App.xaml`

Reply:
@karenbtlai gotcha, sorry didn't understand properly :)

Reply:
We want this only when the new styles are used (when running on any version of the OS). Some hacky solutions 
1. We could use a ThemeResource and read and use in code
2. a Tag in one of the elements in the template
3. Add a column definition to the grid inside the SplitPane and use its width internally.

None of these are ideal, pick one and document it so it is clear why it was done that way.

Reply:
The calculation of newButtonWidths above should have originally been exposed as a template setting, so that this button property would be customizable. Your hitting this issue with the new style be cause that step was skipped. I think we should expose that as a template setting to be used with the old style and expose (newButtonWidths - 8) as a new template setting to be used with the updated style.

Reply:
It's possible to create a shared to function to read s_tlsUseLatestStyle.

Also we need to make sure we didn't break when dictionary is using compact

Reply:
@ranjeshj @YuliKl created new template setting properties, this will need to go under API reviews.